### PR TITLE
Protect font initialization with mutex

### LINKED ISSUE #4532: Changing font settings with /set segfaults
I searched only a little for a pre-existing issue.  Typing `/set font_size 20` crashes minetest and there's no error in debug.txt.  I haven't tried any other settings, just that one string.

### PR DESCRIPTION
**Pull request by sfan5**
fixes #4532

## To do

This PR is a Ready for Review.

### CODE REVIEW

File: src/client/fontengine.cpp [hunk 1]
```diff
@@ -149,42 +153,23 @@ irr::gui::IGUIFont *FontEngine::getFont(FontSpec spec, bool may_fail)
 /******************************************************************************/
 unsigned int FontEngine::getTextHeight(const FontSpec &spec)
 {
-	irr::gui::IGUIFont *font = getFont(spec);
-
-	// use current skin font as fallback
-	if (font == NULL) {
-		font = m_env->getSkin()->getFont();
-	}
-	FATAL_ERROR_IF(font == NULL, "Could not get skin font");
```
Comment (sfan5) [inline src/client/fontengine.cpp:158]: This and other code that assumes `font` can be null can go, `getFont()` *never* returns null

File: src/client/fontengine.cpp [hunk 2]
```diff
@@ -238,22 +223,9 @@ void FontEngine::readSettings()
...
-				"\n\trequired for current screen configuration was not found" <<
-				" or was invalid file format." <<
-				"\n\tUsing irrlicht default font." << std::endl;
-
-	// If we did fail to create a font our own make irrlicht find a default one
-	font = m_env->getSkin()->getFont();
-	FATAL_ERROR_IF(font == NULL, "Could not create/get font");
-
-	u32 text_height = font->getDimension(L"Hello, world!").Height;
-	infostream << "FontEngine: measured text_height=" << text_height << std::endl;
```
Comment (sfan5) [inline src/client/fontengine.cpp:256]: removing the usage of the font here is part of the fix

**Comment by yamanq** [comment]
Tested with a few font settings on a singleplayer game and they all worked without error.

**Review by SmallJoker — APPROVED** [review]
LGTM

**Pull request merged**